dataverse/webapi/C#-NETx/BulkOperations/Utility.cs

```diff
@@ -272,8 +288,186 @@ public static async Task DeleteExampleTable(Service service, string tableSchemaN
+                if (waitForKeyStatus)
+                {
+                    int timeOutInterval = 305; // in seconds //
+                    int pollingInterval = 15;
+                    while (timeOutInterval > 0)
+                    {
+                        Thread.Sleep(pollingInterval * 1000);
+
+                        // Retrieve Key Status
+                        // TODO: WebAPIService.Metadata.Messages should have RetrieveEntityKeyRequest/RetrieveEntityKeyResponse classes
+                        // That align with the SDK classes so this operation is reusable and uses
+                        // the EntityKeyMetadata class we already have
+                        // https://learn.microsoft.com/en-us/dotnet/api/microsoft.xrm.sdk.messages.retrieveentitykeyrequest?view=dataverse-sdk-latest
+                        // https://learn.microsoft.com/en-us/dotnet/api/microsoft.xrm.sdk.messages.retrieveentitykeyresponse?view=dataverse-sdk-latest
+                        var keyMetadata = await GetEntityRecords(service, keyUrl).ConfigureAwait(false);
+
+                        var indexStatus = keyMetadata.Value<string>("EntityKeyIndexStatus");
+
+                        if (indexStatus != null && indexStatus.Equals("Active", StringComparison.OrdinalIgnoreCase))
+                        {
+                            return "Success";
+                        }
+                        else if (indexStatus != null && indexStatus.Equals("Failed", StringComparison.OrdinalIgnoreCase))
+                        {
+                            var asyncJobId = keyMetadata["AsyncJob"].ToString();
 
+                            var asyncJobResponse = await GetEntityRecords(service, "asyncoperations(" + asyncJobId + ")").ConfigureAwait(false);
+
+                            return asyncJobResponse["Message"].ToString();
+                        }
+                        timeOutInterval -= pollingInterval;
+                    }
+                }
+                else
+                {
+                    waitForKeyStatus = true;
+                }
+                return "Pending";
+            }
+            catch (Exception ex)
+            {
+                return ex.Message;
+            }
+        }
+
+        /// <summary>
+		/// Returns the entity records
+		/// </summary>
+		/// <example>accounts</example>
+		/// <example>tasks(6b5941dd-d175-e511-80d4-00155d2a68d1)</example>
+		/// <param name="query">The relative URL of the message request</param>
+		/// <param name="httpRequestHeaders">Http request header details</param>
+		/// <returns>Returns the response object</returns>
+		public static async Task<JObject> GetEntityRecords(
+            Service service, 
+            string query, 
+            Dictionary<string, string> httpRequestHeaders = null)
+        {
+            var request = new RetrieveMultipleRequest(query);
+
+            // TODO: You can use service.SendAsync<RetrieveMultipleResponse>(request) to get 
```

Comment:
You can use service.SendAsync<RetrieveMultipleResponse>(request) to get a typed RetrieveMultipleResponse object returned with a Records property that could simplify this